src/GeoJSON.js

```diff
@@ -0,0 +1,83 @@
+import {Transform} from 'vega-dataflow';
+import {error, inherits, isArray} from 'vega-util';
+
+/**
+ * Convert an array of longitude/latitude points or the collection of the specified field
+ * into GeoJSON for use in a projection's fit argument.
+ * @constructor
+ * @param {object} params - The parameters for this operator.
+ * @param {Array<function(object): *>} params.fields - A two-element array of
+ *   field accessors for the longitude and latitude values.
+ */
+export default function GeoJSON(params) {
+  Transform.call(this, null, params);
+}
+
+export function collectGeoJSON(features) {
+  return !isArray(features) ? features
+    : features.length > 1 ? {type: 'FeatureCollection', features: features}
+    : features[0];
+}
+
+GeoJSON.Definition = {
+  "type": "GeoJSON",
+  "metadata": {},
+  "params": [
+    { "name": "fields", "type": "field", "array": true, "length": 2 },
+    { "name": "geojson", "type": "field" },
+  ]
+};
+
+var prototype = inherits(GeoJSON, Transform);
+
+prototype.transform = function(_, pulse) {
+  var fields = _.fields,
+      lon = fields[0],
+      lat = fields[1],
+      geojson = _.geojson,
+      flag = pulse.ADD,
+      features = [],
```

Comment:
UPDATE: the comments above only apply if you modify the use of `pulse.changed` below. Right now `mod` will just about always be `true` as `pulse.changed()` (with no arguments) returns true if there any changes whatsoever (add/rem/mod). So currently the code resets on every invocation and the rest of the mod checks are superfluous. I think you want to reset if (a) there are removed tuples (`pulse.changed(pulse.REM)`) or (b) there were changes to the accessor functions or their fields. Otherwise, you can perform a more targetted streaming update.